### PR TITLE
Demo for dropwizard-pac4j and jax-rs-pac4j

### PR DESCRIPTION
**Pull request by victornoel**
For now only to protect views and not global filters

@leleuj: not sure why but I don't have the right to push on pac4j/dropwizard-pac4j-demo...

Anyway, here is a first shot at a demo fro `dropwizard-pac4j` (and `jax-rs-pac4j`, not sure I will make a demo also for `jax-rs-pac4j` alone btw...).

### CODE REVIEW

**Comment by leleuj** [comment]
There was an issue in the rights affected to the pac4j developers. It should be ok now.

If you don't need the jax-rs-pac4j-demo, just let me know, I'll remove it.

File: src/main/java/org/pac4j/demo/dw/Pac4JDemoApplication.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import io.dropwizard.setup.Environment;
+import io.dropwizard.views.ViewBundle;
+
+public class Pac4JDemoApplication extends Application<Pac4JDemoConfiguration> {
+
+    public static void main(String[] args) throws Exception {
+        new Pac4JDemoApplication().run(args);
+    }
+
+    private final Pac4jBundle<Pac4JDemoConfiguration> pac4j = new Pac4jBundle<Pac4JDemoConfiguration>() {
```
Comment (leleuj) [inline src/main/java/org/pac4j/demo/dw/Pac4JDemoApplication.java]: Shouldn't this private class available by default in the `dropwizard-pac4j` library?

File: src/main/java/org/pac4j/demo/dw/Pac4JDemoConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package org.pac4j.demo.dw;
+
+import org.pac4j.dropwizard.Pac4jFactory;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import io.dropwizard.Configuration;
+
+public class Pac4JDemoConfiguration extends Configuration {
```
Comment (leleuj) [inline src/main/java/org/pac4j/demo/dw/Pac4JDemoConfiguration.java]: as well as this one?

**Comment by leleuj** [comment]
Two comments.

Merging.

Ping me when you'll feel the demo is ready for more intensive testing from my part.

File: src/main/java/org/pac4j/demo/dw/Pac4JDemoApplication.java (hunk 1 above)
Comment (victornoel) [inline src/main/java/org/pac4j/demo/dw/Pac4JDemoApplication.java]: I don't understand what you mean :)

This code is specific to the demo application, each application must have its own anonymous subclassing of `Pac4jBundle` with respect to their own subclassing of `Configuration`.

File: src/main/java/org/pac4j/demo/dw/Pac4JDemoConfiguration.java (hunk 2 above)
Comment (victornoel) [inline src/main/java/org/pac4j/demo/dw/Pac4JDemoConfiguration.java]: Each dropwizard application needs its own subclassing of Configuration, even if for simple cases they can be similar, most of the time, they will be different, see for example my configuration in an application I'm working on: [here](https://gitlab.com/victornoel/petals-cockpit/blob/master/backend/src/main/java/org/ow2/petals/cockpit/server/configuration/CockpitConfiguration.java).

**Comment by victornoel** [comment]
@leleuj ok, I think I will introduce more examples of how to exploits dropwizard-pac4j and then I will tell you :)

File: src/main/java/org/pac4j/demo/dw/Pac4JDemoApplication.java (hunk 1 above)
Comment (leleuj) [inline src/main/java/org/pac4j/demo/dw/Pac4JDemoApplication.java]: Yes, forget about it.

File: src/main/java/org/pac4j/demo/dw/Pac4JDemoConfiguration.java (hunk 2 above)
Comment (leleuj) [inline src/main/java/org/pac4j/demo/dw/Pac4JDemoConfiguration.java]: OK

**Pull request merged**